src/wallet/rpc/wallet.cpp

```diff
@@ -730,7 +730,9 @@ static RPCHelpMan migratewallet()
             std::shared_ptr<CWallet> wallet = GetWalletForJSONRPCRequest(request);
             if (!wallet) return NullUniValue;
 
-            EnsureWalletIsUnlocked(*wallet);
+            if (wallet->IsCrypted()) {
+                throw JSONRPCError(RPC_WALLET_UNLOCK_NEEDED, "Error: migratewallet on encrypted wallets is currently unsupported.");
```

Comment:
As stated earlier in this PR. The encrypted legacy wallet cannot be migrated even if it is unlocked.
So, in this case, I don't think **RPC_WALLET_UNLOCK_NEEDED** would be an appropriate error code to use.

Suggestion:

We can add a new error code for this particular behavior in the `src/rpc/protocol.h`

```
RPC_WALLET_LEGACY_ENCRYPTED     = -37, //!< The legacy wallet is encrypted and is not yet supported for migration.

Reply:
I agree that `RPC_WALLET_UNLOCK_NEEDED` can be confusing feedback, but I'd prefer not to add an error code for a very specific situation that IIUC is only meant to be temporary. I think we have a couple of existing codes that work fine here, with `RPC_WALLET_WRONG_ENC_STATE` looking like the most suitable: 

```
RPC_WALLET_WRONG_ENC_STATE      = -15, //!< Command given in wrong wallet encryption state (encrypting an encrypted wallet etc.)
```

Reply:
Changed to `RPC_WALLET_WRONG_ENC_STATE`.